Resources/Prototypes/Entities/Structures/Machines/plc.yml

```diff
@@ -0,0 +1,47 @@
+- type: entity
+  id: PLC
+  name: plc
+  description: It's a programmable logic controller.
```

Comment:
This should inherit from BaseStructure. Transform, Clickable, TnteractionOutline, Damageable will all be unnecessary as they are defined in BaseStructure.

Alternatively, you could create a machine board for this and make it inherit from BaseMachine - in that case you'll need to look into how other machines (e.g. autolathe) do it.

Reply:
My intention is to keep it simple.
How do I inherit from BaseStructure? Do I just remove those types under components and replace them with one BaseStructure type?

Reply:
Uh, add `parent: BaseStructure`. Then navigate to the BaseStructure type, look what components it already has that were re-defined in your type, and remove them. If the parent entity defines a component but does not define all the fields you want, you can still add the same component and only set up the fields that were not set up in the parent entity. In that case, The rest of the fields will be inherited from the parent.